tests/Annotations/Command/QueueWatchCommandTest.php

```diff
@@ -142,6 +162,36 @@ public function providerProfiles()
                 'email' => 'username@hypothesis.elifesciences.org',
                 'display_name' => 'This display name is way too l',
             ],
+            [
+                [LogLevel::INFO, 'The display name for profile "username" is too long and has been truncated from "This display name is way too long" to "This display name is way too l".', []],
+                [LogLevel::INFO, 'No email address for profile "username", backup email address created.', []],
+            ],
+        ];
+        yield 'with single email' => [
+            new InternalSqsMessage('profile', 'username'),
+            new Profile('username', new PersonDetails('PreferredName', 'IndexName'), new EmptySequence(), new ArraySequence(['username@email.com'])),
+            [
+                'username' => 'username',
+                'email' => 'username@email.com',
+                'display_name' => 'PreferredName',
+            ],
+            [],
+            [
```

Comment:
While the whitelist seems very precise in specifying what has to be logged, the blacklist is more tricky as the specified entry could very well be logged with a different level, or with a very slightly different message (e.g. without the final `.`)

Perhaps substitute the blacklist with an assertion that *all* log messages must be in the whitelist, otherwise they are not accounted for and fail the test?

Reply:
Design-wise, in cases where we need this level of introspection for what is happening an Observer pattern can come up, but let's not make it more complex. I suspect the logic in the command will soon grow and something will be extracted from it.

Reply:
Having reviewed these blacklist checks, they add no value so I have removed them.